File: android/app/src/main/java/com/xaymaca/sit/ui/warm/MonogramAvatar.kt
package com.xaymaca.sit.ui.warm

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.PhotoCamera
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.PathEffect
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.xaymaca.sit.ui.theme.WarmCategory

/** Circular avatar with initials, tinted to the category accent. List row size. */
@Composable
fun MonogramAvatar(
    initials: String,
    category: WarmCategory,
    modifier: Modifier = Modifier,
    size: Dp = 36.dp,
) {
    val palette = category.palette
    Box(
        modifier = modifier
            .size(size)
            .clip(CircleShape)
            .background(palette.accent),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            text = initials.uppercase(),
            style = TextStyle(
                fontSize = (size.value * 0.40f).sp,
                fontWeight = FontWeight.SemiBold,
                color = Color(0xFFFAF4E2),
            ),
        )
    }
}

/**
 * Empty-state photo affordance for the Contact Detail header. Dashed
 * circle in the category accent + faded monogram + camera badge.
 * Spec: HANDOFF.md lines 22-30.
 */
@Composable
fun MonogramPhotoAffordance(
    initials: String,
    category: WarmCategory,
    modifier: Modifier = Modifier,
    size: Dp = 132.dp,
) {
    val palette = category.palette
    Box(
        modifier = modifier.size(size),
        contentAlignment = Alignment.Center,
    ) {
        // Dashed circle border
        Canvas(modifier = Modifier.size(size)) {
            val strokeWidth = 2f
            drawCircle(
                color = palette.accent.copy(alpha = 0.55f),
                radius = (size.toPx() - strokeWidth) / 2f,
                center = Offset(this.size.width / 2f, this.size.height / 2f),
                style = Stroke(
                    width = strokeWidth,
                    pathEffect = PathEffect.dashPathEffect(floatArrayOf(8f, 6f), 0f),
                ),
            )
        }
        // Faded monogram
        Text(
            text = initials.uppercase(),
            style = TextStyle(
                fontSize = (size.value * 0.32f).sp,
                fontWeight = FontWeight.SemiBold,
                color = palette.accent.copy(alpha = 0.32f),
            ),
        )
        // Camera badge bottom-right
        val badgeSize = size * 0.27f
        Box(
            modifier = Modifier
                .align(Alignment.BottomEnd)
                .offset(x = size * 0.03f, y = size * 0.03f)
                .size(badgeSize)
                .clip(CircleShape)
                .background(palette.accent),
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                imageVector = Icons.Filled.PhotoCamera,
                contentDescription = null,
                tint = Color.White,
                modifier = Modifier.size(badgeSize * 0.5f),
            )
        }
    }
}

File: ios/Sources/SIT/Views/Shared/Warm/MonogramAvatar.swift
import SwiftUI

/// Circular avatar showing a contact's initials, tinted by their category.
/// Two callers: list rows (`size: 36`) and the empty contact-detail
/// affordance (`size: 132, dashed: true`). For the dashed variant + camera
/// badge + "Add a photo" link, see `MonogramPhotoAffordance`.
struct MonogramAvatar: View {
    let initials: String
    let category: WarmCategory
    var size: CGFloat = 36

    var body: some View {
        let palette = category.palette
        Circle()
            .fill(palette.accent)
            .frame(width: size, height: size)
            .overlay(
                Text(initials.uppercased())
                    .font(.system(size: size * 0.40, weight: .semibold))
                    .foregroundStyle(Color(red: 0.98, green: 0.96, blue: 0.89))
            )
    }
}

/// Empty-state photo affordance for the Contact Detail header.
/// Dashed circle in the category accent + faded monogram + camera badge.
/// Spec: HANDOFF.md lines 22-30.
struct MonogramPhotoAffordance: View {
    let initials: String
    let category: WarmCategory
    var size: CGFloat = 132

    var body: some View {
        let palette = category.palette
        ZStack(alignment: .bottomTrailing) {
            ZStack {
                Circle()
                    .strokeBorder(
                        palette.accent.opacity(0.55),
                        style: StrokeStyle(lineWidth: 2, dash: [8, 6])
                    )
                    .frame(width: size, height: size)
                Text(initials.uppercased())
                    .font(.system(size: size * 0.32, weight: .semibold))
                    .foregroundStyle(palette.accent.opacity(0.32))
            }

            Circle()
                .fill(palette.accent)
                .frame(width: size * 0.27, height: size * 0.27)
                .overlay(
                    Image(systemName: "camera.fill")
                        .font(.system(size: size * 0.12, weight: .semibold))
                        .foregroundStyle(.white)
                )
                .offset(x: size * 0.03, y: size * 0.03)
        }
        .frame(width: size, height: size)
    }
}

#Preview("List avatars") {
    HStack(spacing: 16) {
        ForEach(WarmCategory.allCases) { cat in
            MonogramAvatar(initials: "AB", category: cat)
        }
    }
    .padding()
    .background(WarmTheme.subtle.paper)
}

#Preview("Photo affordance") {
    HStack(spacing: 24) {
        ForEach([WarmCategory.family, .friends, .work]) { cat in
            MonogramPhotoAffordance(initials: "JS", category: cat)
        }
    }
    .padding()
    .background(WarmTheme.subtle.paper)
}
